lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -318,16 +318,16 @@ struct ConvertLayoutOpUsingLinearLayoutsConversion
                                  kBlock));
 
     assert(!cvtNeedsSharedMemory(op.getSrc().getType(), op.getType()));
-    assert(ArrayRef(to_vector(dstToSrc->getInDimNames())) ==
-           ArrayRef{kRegister});
     assert(ArrayRef(to_vector(dstToSrc->getOutDimNames())) ==
            ArrayRef{kRegister});
+    assert(dstToSrc->sublayoutIsZero({kLane, kWarp, kBlock}, {kRegister}));
+    auto subLayout = dstToSrc->sublayout({kRegister}, {kRegister});
 
     auto inVals = unpackLLElements(loc, adaptor.getSrc(), rewriter);
     SmallVector<Value> outVals;
-    outVals.resize(dstToSrc->getInDimSize(kRegister));
-    for (int i = 0; i < dstToSrc->getInDimSize(kRegister); i++) {
-      auto srcIdx = dstToSrc->apply({{kRegister, i}});
+    outVals.resize(subLayout.getInDimSize(kRegister));
+    for (int i = 0; i < subLayout.getInDimSize(kRegister); i++) {
+      auto srcIdx = subLayout.apply({{kRegister, i}});
```

Comment:
It couldn't because the following condition doesn't return true any more:

`assert(ArrayRef(to_vector(dstToSrc->getInDimNames())) ==`

We must get a subLayout first.

Reply:
Perhaps worth writing a comment?  "You might be tempted to do X, but it doesn't work because Y."?